ledger/daml-on-sql/README.rst

```diff
@@ -0,0 +1,703 @@
+
+This is because DAML-on-SQL manages its own database schema,
+applying migrations if necessary when upgrading versions.
+
+To specify the PostgreSQL instance you wish to connect, use the
+``--sql-backend-jdbcurl <value>`` command line option, where ``<value>``
+is a valid JDBC URL containing the username, password and database
+name to connect to.
+
+Here is an example for such a URL: ``jdbc:postgresql://localhost/test?user=fred&password=secret``
+
+Due to possible conflicts between the ``&`` character and various terminal
+shells, we recommend quoting the JDBC URL like so:
+
+.. code-block:: none
+
+  $ java -jar daml-on-sql-<version>.jar Main.dar --sql-backend-jdbcurl 'jdbc:postgresql://localhost/test?user=fred&password=secret'
+
+If you're not familiar with JDBC URLs, see the `PostgreSQL JDBC docs for more information <https://jdbc.postgresql.org/documentation/head/connect.html>`__.
+
+Architecture and availability
+*****************************
+
+Processes and components
+========================
+
+The core processes necessary to run a DAML-on-SQL deployment are:
+
+- the JVM-hosted standalone Ledger API server and
+- the PostgreSQL server used to persist the ledger data.
+
+The Ledger API server communicates with the external world via the gRPC
+Ledger API and communicates with PostgreSQL via JDBC to persist transactions,
+keep track of active contracts, store compiled DAML packages, and so on.
+
+Core architecture considerations
+================================
+
+A very important point to make is that the backing PostgreSQL server performs a
+lot of work which is both CPU- and IO-intensive: all (valid) Ledger API requests
+will eventually hit the database. At the same time, the Ledger API server has to
+have available resources to validate requests, evaluate commands and prepare responses.
+While the PostgreSQL schema is designed to be as efficient as possible, practical
+experience has shown that having **dedicated computation and memory resources for the
+two core components** (the Ledger API server and the PostgreSQL server) allows the two
+to run interfering as little as possible with each other. Depending on the kind of
+deployment you wish to make, this can be achieved with containerization, virtualization
+or simply using physically different machines. Still, the Ledger API communicates
+abundantly with the PostgreSQL server and many Ledger API requests need to go all
+the way to persist information on the database. To reduce the latency necessary to
+serve outstanding requests, **the Ledger API server and PostgreSQL server should be
+physically co-located**.
+
+Core availability considerations
+================================
+
+In order to address availability concerns, it's important to understand what each of the
+core components do and how they interact with each other, in particular regarding state
+and consistency.
+
+In DAML-on-SQL, the Ledger API server holds important state that affects the offset
```

Comment:
I'm not sure the details about why a multi-participant deployment for DAML on SQL doesn't work are necessary to convey that only a single DAML on SQL process should be connected at the same time to a particular postgres schema.

Reply:
I'm not telling any detail. I'm just providing a relatively vague motivation (state-dependent behavior) regarding why it's important to have a single DAML on SQL instances running atop a single PostgreSQL database. I can remove the reasoning, but I'm not sure that would effectively tell users that there are inherent dangers in ending up with two active DAML on SQL instances interacting with a single PostgreSQL installation.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6850/commits/f57199d0811464e1f0c13d7b9bd3a98f5873cd7a